src/core/nearfieldreader.cpp

```diff
@@ -0,0 +1,144 @@
+/***************************************************************************
+ nearfieldreader.cpp - NearFieldReader
+
+ ---------------------
+ begin                : 27.08.2023
+ copyright            : (C) 2023 by Mathieu Pellerin
+ email                : mathieu (at) opengis.ch
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "nearfieldreader.h"
+
+#include <QDebug>
+#include <QNdefMessage>
+#include <QNdefNfcTextRecord>
+#include <QNearFieldTarget>
+#include <QUrl>
+#include <qgsmessagelog.h>
+
+NearFieldReader::NearFieldReader( QObject *parent )
+  : QObject( parent )
+{
+  mNearFieldManager = std::make_unique<QNearFieldManager>( this );
+  connect( mNearFieldManager.get(), &QNearFieldManager::targetDetected, this, &NearFieldReader::handleTargetDetected );
+  connect( mNearFieldManager.get(), &QNearFieldManager::targetDetected, this, &NearFieldReader::handleTargetLost );
+}
+
+NearFieldReader::~NearFieldReader()
+{
+#if QT_VERSION < QT_VERSION_CHECK( 6, 0, 0 )
+  mNearFieldManager->setTargetAccessModes( QNearFieldManager::NoTargetAccess );
```

Comment:
Can you add a comment why we need that on destruction?

Reply:
@m-kuhn , I think we can skip it actually, removed.